dango/types/src/account_factory/msg.rs

```diff
@@ -75,13 +73,13 @@ pub enum QueryMsg {
         start_after: Option<Addr>,
         limit: Option<u32>,
     },
-    /// Query a key by its hash.
+    /// Query a key by its hash associated to a username.
     #[returns(Key)]
-    Key { hash: Hash160 },
-    /// Enumerate all keys and hashes.
-    #[returns(BTreeMap<Hash160, Key>)]
+    Key { hash: Hash160, username: Username },
+    /// Enumerate all keys.
+    #[returns(BTreeMap<(Username, Hash160), Key>)]
```

Comment:
> `BTreeMap<(Username, Hash160), Key>`

I don't think this return type would work, because you can't serialize it to JSON. You will get "key must be a string" error when you to `.to_json_value`.

We can do this:

```rust
pub struct QueryKeyResponseItem {
    pub username: Username,
    pub key_hash: Hash160,
    pub key: Key,
}
```

Then use `Vec<QueryKeyResponseItem>` as the return type.